PostBy/View Models/PostViewModel.m

```diff
@@ -125,29 +125,103 @@ - (void) checkIfDisliked {
 }
 
 - (void) likeButtonTap {
-    if (self.isLiked) {
-        [self unlikePost];
-    } else {
-        [self likePost];
-        if (self.isDisliked) {
-            [self undislikePost];
-        }
-    }
-    
-    [self saveAndRefresh];
+    [self interactWithPostBy:@"like"];
 }
 
 - (void) dislikeButtonTap {
-    if (self.isDisliked) {
-        [self undislikePost];
-    } else {
-        [self dislikePost];
-        if (self.isLiked) {
-            [self unlikePost];
+    [self interactWithPostBy:@"dislike"];
+}
+
+- (void) interactWithPostBy:(NSString *)action {
```

Comment:
Usually we set 40 lines as the threshold to decompose complex logic, to make code clean and avoid others getting lost in the logic.

For this function, you may consider to extract the logic of "Perform action based on local post" to a separate function, so that people can skip understanding the details when they read this "interactWithPostBy" function. If they want to check the detailed logic, they just jump to the separated one.